src/app/services/mocks/tree-node.service.mock.ts

```diff
@@ -16,17 +16,13 @@ export class TreeNodeServiceMock {
   public treeNodes: TreeNode[] = [];
   // the copy of the tree nodes that is used for constructing the tree in the 2nd step (projection)
   public treeNodesCopy: TreeNode[] = [];
-  // the entire tree table data that holds the patients' observations in the 2nd step (projection)
-  private _projectionTreeData: TreeNode[] = [];
-  // the selected tree table data that holds the patients' observations in the 2nd step (projection)
-  private _selectedProjectionTreeData: TreeNode[] = [];
-  // the final tree nodes resulted from data selection
-  private _finalTreeNodes: TreeNode[] = [];
+  private _variablesTreeData: TreeNode[] = [];
 
   public selectedTreeNode;
   public treeNodeCallsSent = 0; // the number of tree-node calls sent
   public treeNodeCallsReceived = 0; // the number of tree-node calls received
   public treeNodesUpdated: Subject<boolean> = new Subject<boolean>();
+  public selectedVariablesTreeDataUpdated: Subject<TreeNode[]> = new Subject<TreeNode[]>();
```

Comment:
Instead of having variable selection update events in both services (`selectedVariablesTreeDataUpdated` in `TreeNodeService` and `selectedVariablesUpdated` in `ConstraintService`), it is better to have one place (possibly in a variable selection service), have all components update the selection there, and having the tree node service and constraint service (or perhaps categorized variable component) listen to these updates to apply the selection.